src/libcore/sync/atomic.rs

```diff
@@ -1166,6 +1166,216 @@ impl<T> AtomicPtr<T> {
             }
         }
     }
+
+    /// Adds to the current pointer, returning the previous pointer.
+    ///
+    /// Unlike other pointer additions, `fetch_add` increments directly by the provided value,
```

Comment:
Why would this divergence from normal pointers be preferable?

Reply:
Ah, sorry, I wrote this in the PR/commit, but not in the docs:
> The rationale for this is that for atomic operations, the user may indeed wish to truly increment by 1, which is difficult if all deltas are interpreted in increments of the type's size.

Reply:
Why would a user of a pointer-that-is-atomic want to increment by 1 and a user of a pointer-that-is-not-atomic not want to increment by 1?